kod/object/item.kod

```diff
@@ -633,6 +689,41 @@ messages:
       return FALSE;
    }
 
+   GetMagicRarityGradeCount()
+   "Checks plItem_attributes with GetRarityCountModifier().  "
+   "Adds each attribute's rarity count to the total.  "
+   "Return either viRarity or count, whichever is larger.  "
+   "This ensures the higher rarity is chosen if in the case "
+   "of competing rarities for an item."
+   {
+
+      local i, iNum, oItemAtt, count;
+      count = 0;
+
+      if Send(self,@HasAnyAttribute)
+      {
+         for i in plItem_attributes
+         {
+            iNum = Send(self,@GetNumFromCompound,#compound=First(i));
+            oItemAtt = Send(SYS,@FindItemAttByNum,#num=iNum);
+            count = count + Send(oItemAtt,@GetRarityCountModifier);
+         }
+      }
+
+      return Bound(count, viRarity, $);
+   }
+
+   GetRarityCountModifier()
+   "Returns a modifier count to an item's rarity grade.  "
+   "Used to modify the rarity grade for an item based on "
+   "notable item attributes.  Positive means increase the "
+   "rarity grade, negative means decrease the rarity grade."
+   {
+      local count;
```

Comment:
Soo.. 
```
return 0;
```
?

Reply:
Changed to return 0 as suggested.  I definitely overthink things or leave relics of my thought process sometimes.  We can modify individual item attributes' `GetRarityCountModifier` messages to affect rarities in the future, but it makes sense for the superclass message of item to default to return 0 (no change).